### PR TITLE
refactor(demo): Site header harmonised - Mandatory/optional knobs - FRONT-918

### PR DESCRIPTION
**Pull request by planctus**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js [hunk 1]
```diff
@@ -41,44 +55,118 @@ const btnLoginHandler = () => {
   }
 };
 
+const prepareSiteHeaderHarmonised = data => {
+  data.group = select('group', [data.group], data.group, buttonLabels.required);
```
Comment (papegaill) [inline src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js]: on G1, the group is _undefined_

File: src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js [hunk 2]
```diff
@@ -41,44 +55,118 @@ const btnLoginHandler = () => {
...
+  }
+
+  return data;
+};
 storiesOf('Components/Site Headers/Harmonised', module)
   .addDecorator(withNotes)
   .addDecorator(withCode)
   .addDecorator(withKnobs)
   .add(
     'group 1',
```
Comment (papegaill) [inline src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js]: Maybe a nice use case to have is:
`with or without menu`
What do you think?

Regarding this I think we have a problem here(more on ecl side), as the menu is optional and the site name mandatory, if I remove the menu, there is something wrong with padding margin.. also the site-name is a child  element of the nav tag. shouldn't it be separated from navigation?

File: src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js (hunk 1 above)
Comment (planctus) [reply src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js]: ;)

File: src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js (hunk 2 above)
Comment (planctus) [reply src/ec/packages/ec-component-site-header-harmonised/site-header-harmonised.story.js]: What the heck...good that you noticed this, not easy to spot it, the only example if i'm not mistaken is in that "event demo" on ECL.
It wasn't too hard to adapt the template, it's more the stories that changed a lot, there is almost  no required element anymore.

**Pull request merged**